demisto_sdk/commands/common/hook_validations/base_validator.py

```diff
@@ -130,3 +132,71 @@ def _is_id_equals_name(self, file_type):
             print_error(Errors.suggest_fix(self.file_path))
             return False
         return True
+
+    def _load_conf_file(self):
+        with open(self.CONF_PATH) as data_file:
+            return json.load(data_file)
+
+    def are_tests_configured(self) -> bool:
+        """
+        Checks if a file (playbook or integration) has a TestPlaybook and if the TestPlaybook is configured in conf.json
+        And prints an error message accordingly
+        """
+        file_type = self.structure_validator.scheme_name
+        tests = self.current_file.get('tests', [])
+        if not self.yml_has_test_key(tests, file_type):
+            return False
+        return self.tests_registered_in_conf_json_file(tests)
+
+    def tests_registered_in_conf_json_file(self, tests):
+        # If 'No tests' is explicitly configured, we can skip checking if the tests are
+        # registered in the conf.json file.
+        no_tests_explicitly = any(test for test in tests if 'no test' in test.lower())
+        if no_tests_explicitly:
+            return True
+        conf_json__tests = self._load_conf_file()['tests']
+
+        # Calling it integration but the validation is the same for integrations, playbooks and scripts
+        integration_id = self._get_file_id(self.structure_validator.scheme_name)
+        not_registered_tests = self.get_not_registered_tests(conf_json__tests, integration_id, tests)
+        if not_registered_tests:
+            missing_test_configurations = json.dumps([
+                {'integrations': integration_id, 'playbookID': test} for test in not_registered_tests
+            ], indent=4).strip('[]')
+            error_message = \
+                f'The following TestPlaybooks are not registered in {self.CONF_PATH} file.\n' \
+                f'Please add\n{missing_test_configurations}\nto {self.CONF_PATH} path under \'tests\' key ' \
+                f'or run \'demisto-sdk format --skip-test-playbooks -p {self.file_path}\''
+            print_error(error_message)
+            return False
+        return True
+
+    def get_not_registered_tests(self, conf_json_data, integration_id, tests):
+        not_registered_tests = []
+        for test in tests:
+            test_registered_in_conf_json = any(
+                test_config for test_config in conf_json_data if self.find_test_match(test_config, test, integration_id)
+            )
+            if not test_registered_in_conf_json:
+                not_registered_tests.append(test)
+        return not_registered_tests
+
+    def yml_has_test_key(self, tests, file_type):
+        if not tests:
+            print_error(
+                f'You don\'t have a TestPlaybook for {file_type} {self.file_path}. If you don\'t want to create a'
+                f' TestPlaybook for this {file_type}, edit the yml file and add  \ntests:\n -  No tests\n lines'
+                f' to it. Or run demisto-sdk format --skip-test-playbooks -p {file_type}')
```

Comment:
This error message isn't entirely accurate. They might have created a test playbook but it simply has not been linked to the content item. Please update the message to reflect that.

Reply:
We shouldn't only provide the easy way out (aka by telling them about the option to add `no tests`). You need to do one of two things. 
1. **Less Ideal Option**: Before telling them how they can avoid having a test playbook, describe that they should create a test playbook and then add the ID of the test playbook to the `tests` key in the YAML of the content item - similar to how you described how they could add `No tests` themselves.
2. **More Ideal Option**: Add something like an `--attach-test-playbook` option to the `format` command where the user would give the paths to the content item and the test playbook and the command would add the ID of the test playbook to the `tests` key of the content item's YAML. Then you would describe this option first in your error message.

Given time constraints, we may have to go with the less ideal option.

Reply:
The flag is not implemented currently anyways, so I'll remove the flag part until it is.